### PR TITLE
Review 1

### PR DESCRIPTION
**Pull request by duythien**
Strict warning: call_user_func() expects parameter 1 to be a valid callback, non-static method getInfo()

Test case #191

### CODE REVIEW

File: tests/atest_base/atest_base.module [hunk 1]
```diff
@@ -38,6 +38,16 @@ class At_Base_Test_Class {
   public static function staticGetVariables() {
     return atest_variables();
   }
+
+  /**
+   * Function test case Render content with class
+   * @param  string $argument 
+   * @return 
+   */
+  public function bar($argument=''){
```
Comment (and1truong) [inline tests/atest_base/atest_base.module:47]: ){ => ) {

File: tests/atest_base/atest_base.info [hunk 2]
```diff
@@ -2,7 +2,7 @@ name = ATest Base
 description = Make sure at_base.module is working
 package = Testing
 core = 7.x
-hidden = TRUE
+hidden = FALSE
```
Comment (and1truong) [inline tests/atest_base/atest_base.info:5]: Why?

**Comment by and1truong** [comment]
One question. What are you doing here. What do you mean for "Review 1"?

File: lib/Config/Config.php [hunk 3]
```diff
@@ -126,4 +126,52 @@ public function setAll($data) {
...
+    }
+
+    if (!function_exists('spyc_load_file')) {
+      require_once DRUPAL_ROOT . '/sites/all/libraries/spyc/Spyc.php';
+    }
+
+    //return \Spyc::YAMLLoad($path);
+    return spyc_load_file($path);
+  }
 }
```
Comment (and1truong) [inline lib/Config/Config.php:177]: Please add new methods to Drupal\at_base\Config\Resolver instead.

File: lib/Config/Config.php [hunk 4]
```diff
@@ -126,4 +126,52 @@ public function setAll($data) {
...
+  /**
+   * Read YAML file.
+   *
+   * @param  string $path Path to yaml file.
+   * @return mixed
+   */
+  public function read_yaml($path) {
+    
+    if (!is_file(DRUPAL_ROOT . '/sites/all/libraries/spyc/Spyc.php')) {
+      throw new \RuntimeException('Missing library: spyc');
```
Comment (and1truong) [inline lib/Config/Config.php:167]: yaml_parse() function is already available https://github.com/atdrupal/at_base/blob/dev-master/at_base.hack.inc#L27

File: lib/Config/Config.php [hunk 5]
```diff
@@ -126,4 +126,52 @@ public function setAll($data) {
...
+    return $this->read_yaml($dir_file);
+  }
+
+  /**
+   * Read YAML file.
+   *
+   * @param  string $path Path to yaml file.
+   * @return mixed
+   */
+  public function read_yaml($path) {
```
Comment (and1truong) [inline lib/Config/Config.php:164]: Do not need method. Use yaml_parse() directly https://github.com/atdrupal/at_base/blob/dev-master/at_base.hack.inc#L27

File: tests/atest_config/config/config.yml [hunk 6]
```diff
@@ -1,3 +1,8 @@
 foo: bar
 structured_data:
-  foo: bar
+  foo: helllo
+
+routes:
+  hello:
+    @include: base.yml
```
Comment (and1truong) [inline tests/atest_config/config/config.yml:7]: Please also test on nested tree — @include: sub_dir/base.yml

File: tests/atest_base/atest_base.module [hunk 7]
```diff
@@ -38,6 +38,16 @@ class At_Base_Test_Class {
   public static function staticGetVariables() {
     return atest_variables();
   }
+
+  /**
+   * Function test case Render content with class
+   * @param  string $argument 
+   * @return
```
Comment (and1truong) [inline tests/atest_base/atest_base.module:45]: @return string

File: lib/Helper/Content_Render.php [hunk 8]
```diff
@@ -66,6 +66,8 @@ public function render($data = NULL) {
       $this->setdata($data);
     }
 
+    //var_dump($this->build());
```
Comment (and1truong) [inline lib/Helper/Content_Render.php:69]: Remove

File: lib/Drupal/at_base/Tests/Unit/ServiceContentRenderTest.php [hunk 9]
```diff
@@ -74,4 +74,51 @@ public function testDynamicVariables() {
     $data['controller'] = array('At_Base_Test_Class', 'hi');
     $this->assertEqual($expected, $this->render->render($data));
   }
+
+  /**
+   * Conten render Support prefix, suffix
+   */
+  public function testPrefixSuffix() {
```
Comment (and1truong) [inline lib/Drupal/at_base/Tests/Unit/ServiceContentRenderTest.php:81]: This PR for prefix/suffix or this is about supporting @include in at_config()?

File: lib/Config/Config.php [hunk 10]
```diff
@@ -126,4 +126,52 @@ public function setAll($data) {
...
+    
+    if (!is_file(DRUPAL_ROOT . '/sites/all/libraries/spyc/Spyc.php')) {
+      throw new \RuntimeException('Missing library: spyc');
+    }
+
+    if (!function_exists('spyc_load_file')) {
+      require_once DRUPAL_ROOT . '/sites/all/libraries/spyc/Spyc.php';
+    }
+
+    //return \Spyc::YAMLLoad($path);
```
Comment (and1truong) [inline lib/Config/Config.php:174]: Remove

File: lib/Config/Config.php [hunk 11]
```diff
@@ -126,4 +126,52 @@ public function setAll($data) {
...
+    $dir_file = drupal_get_path('module', $this->module).'/config/'.$filename;
+    
+    return $this->read_yaml($dir_file);
+  }
+
+  /**
+   * Read YAML file.
+   *
+   * @param  string $path Path to yaml file.
+   * @return mixed
```
Comment (and1truong) [inline lib/Config/Config.php:162]: @return array

File: lib/Config/Config.php (hunk 4 above)
Comment (duythien) [inline lib/Config/Config.php:167]: How to call yaml_parse() function in file   lib/Config/Config.php

Comment (and1truong) [inline lib/Config/Config.php:167]: It's always available, just call it like any other function call.

File: lib/Config/Config.php (hunk 10 above)
Comment (duythien) [inline lib/Config/Config.php:174]: ok

File: tests/atest_config/config/config.yml (hunk 6 above)
Comment (duythien) [inline tests/atest_config/config/config.yml:7]: It working

**Pull request closed without merging**